### PR TITLE
Improve admin (SH-494)

### PR DESCRIPTION
**Pull request by Pikkupomo**
Also various fixes

SH-494

### CODE REVIEW

File: shuup/admin/static_src/base/js/drag-n-drop.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    var sortable = Sortable.create(el, {
+        handle: ".sorting-handle",
+        onEnd: function (/**Event*/evt) {
+            // evt.oldIndex;  // element's old index within parent
+            // evt.newIndex;  // element's new index within parent
+            updateOrdering();
+        },
+    });
+    updateOrdering();
+    $(document).on("click", ".btn-remove-sortable", removeFromActive);
```
Comment (gurch101) [inline shuup/admin/static_src/base/js/drag-n-drop.js]: This entire file introduces a whole bunch of global functions that aren't used anywhere other than view settings. This should be put into the extra_js of the template. At the very least, wrap this in a IIFE.

File: shuup/admin/static_src/base/js/drag-n-drop.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    var $target = $("#" + targetElement);
+    var $source = $("#" + sourceElement);
+    window.targetElement = $target;
+    window.sourceElement = $source;
+
+    var el = document.getElementById(targetElement);
+    var sortable = Sortable.create(el, {
+        handle: ".sorting-handle",
+        onEnd: function (/**Event*/evt) {
+            // evt.oldIndex;  // element's old index within parent
```
Comment (gurch101) [inline shuup/admin/static_src/base/js/drag-n-drop.js]: remove commented code

File: shuup/admin/static_src/base/js/main-menu.js [hunk 3]
```diff
@@ -71,4 +71,16 @@ $(function() {
...
+    if (window.Masonry) {
+        const Masonry = window.Masonry;
+        $(".category-menu-content").each(function(idx, elem) {
+            const msnry = new Masonry(elem, {
+                itemSelector: '.submenu-container',
+                columnWidth: '.submenu-container',
+                percentPosition: true
+            });
+        });
+        // console.log(document.getElementsByClassName("category-menu-content"));
```
Comment (gurch101) [inline shuup/admin/static_src/base/js/main-menu.js]: remove this

File: shuup/admin/static_src/base/js/slugify.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * This file is part of Shuup.
+ *
+ * Copyright (c) 2012-2016, Shoop Commerce Ltd. All rights reserved.
+ *
+ * This source code is licensed under the AGPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// https://gist.github.com/mathewbyrne/1280286
+function slugify(text) {
```
Comment (gurch101) [inline shuup/admin/static_src/base/js/slugify.js]: put these two functions in the function expression below.

File: shuup/admin/static_src/base/less/shuup/auth.less [hunk 5]
```diff
@@ -10,7 +10,7 @@
     &-content {
         padding: 30px;
         background: #fff;
-        border-radius: 4px;
+        //border-radius: 4px;
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/auth.less]: remove comment

File: shuup/admin/static_src/base/less/shuup/base.less [hunk 6]
```diff
@@ -1,6 +1,7 @@
 html {
     height: 100%;
     font-size: 62.5%;
+    //font-size:50%;
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/base.less]: remove comment

File: shuup/admin/static_src/base/less/shuup/buttons.less [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.btn {
+  border-radius: 0px;
+  border-width: 1px !important;
+  &.btn-red {
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/buttons.less]: any reason you can't use `btn-danger`?

File: shuup/admin/static_src/base/less/shuup/custom-forms.less [hunk 8]
```diff
@@ -306,7 +306,7 @@ fieldset[disabled] .form-control {
     padding: 10px 16px;
     font-size: 1.6rem;
     line-height: 1.333333;
-    border-radius: 6px;
+    //border-radius: 6px;
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/custom-forms.less]: remove comment

File: shuup/admin/static_src/base/less/shuup/list-settings.less [hunk 9]
```diff
@@ -0,0 +1,39 @@
+.sortable {
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/list-settings.less]: indent this file with 4 spaces

File: shuup/admin/static_src/base/less/shuup/main-menu.less [hunk 10]
```diff
@@ -222,37 +246,35 @@
...
-                    font-size: 1.2rem;
-                }
-            }
-            .content-with-logo {
-                @media (max-width: @screen-sm-min) {
-                    margin-left: 0;
-                }
+    .category-menu-content {
+        h3 {
+            //font-size: 1.6rem;
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/main-menu.less]: remove comment

File: shuup/admin/static_src/base/less/shuup/pagination.less [hunk 11]
```diff
@@ -0,0 +1,46 @@
+.pagination {
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/pagination.less]: remove comments in this file and indent with 4 spaces

File: shuup/admin/static_src/base/less/shuup/picotable.less [hunk 12]
```diff
@@ -5,11 +5,12 @@
             th {
```
Comment (gurch101) [inline shuup/admin/static_src/base/less/shuup/picotable.less]: remove comments in this file

File: shuup/admin/static_src/dashboard/bower.json [hunk 13]
```diff
@@ -8,7 +8,9 @@
   ],
   "dependencies": {
     "chartist": "0.9.5",
-    "masonry": "3.3.2"
+//    "get-size": "^2.0.2",
```
Comment (gurch101) [inline shuup/admin/static_src/dashboard/bower.json]: remove comments

File: shuup/admin/templates/shuup/admin/base/_main_menu.jinja [hunk 14]
```diff
@@ -4,26 +4,55 @@
...
     <a href="#" class="category-menu-close"><i class="fa fa-close"></i></a>
     <div class="category-menu-content">
-        <ul>
-        {% for entry in category %}
-            <li>
-                <a href="{{ entry.url }}">{{ entry.text }}</a>
-            </li>
+        {% for child in category.children %}
+            <div class="submenu-container">
+            <h3 style="color:#fff;">{{ child.name }}</h3>
```
Comment (gurch101) [inline shuup/admin/templates/shuup/admin/base/_main_menu.jinja]: remove inline style

File: shuup/simple_cms/admin_module/__init__.py [hunk 15]
```diff
@@ -43,9 +43,11 @@ def get_help_blocks(self, request, kind):
             yield SimpleHelpBlock(
                 text=_("Add a web page"),
                 actions=[{
-                    "text": _("Add a page"),
+                    "text": _("New CMS page"),
```
Comment (gurch101) [inline shuup/simple_cms/admin_module/__init__.py]: "New page" would be better imo. CMS is meaningless to end users.

File: shuup/admin/modules/settings/__init__.py [hunk 16]
```diff
@@ -36,17 +36,60 @@ def __init__(self, model, default_columns, *args, **kwargs):
         self.default_columns = default_columns
         self.ensure_initial_columns(default_columns)
         self.column_spec = self._build_settings_columns()
-        self.columns = [column for column in self.column_spec if self.get_config(column.id)]
+        self._create_active_columns()
+        self._create_inactive_columns()
+
+    def _create_active_columns(self):
```
Comment (gurch101) [inline shuup/admin/modules/settings/__init__.py]: I don't understand the point of these two functions but 99% of the code in them is similar. Refactor please.

File: shuup/admin/static_src/base/less/shuup/buttons.less (hunk 7 above)
Comment (Pikkupomo) [inline shuup/admin/static_src/base/less/shuup/buttons.less]: I think it's wrong color, I was instructed to use color scheme defined for front page dashboard.

**Comment by tulimaki** [comment]
You really should split this up a little. Maybe do the separation @gurch101 is suggesting. Also the changelog is missing. Please update the translations too.

It also seems that your changes to tests is only fixing the existing ones. You could add new tests around these new features.

**Pull request merged**